app/controllers/miq_policy_controller.rb

```diff
@@ -644,39 +654,38 @@ def replace_right_cell(nodetype, replace_trees = [], presenter = nil)
+                          _("Adding a new Action")
                         else
-                          if @edit
-                            _("Editing %{model} \"%{name}\"") %
-                            {:name  => @action.description.gsub(/'/, "\\\\'"),
-                             :model => ui_lookup(:model => 'MiqAction')}
-                          else
-                            _("%{model} \"%{name}\"") %
-                            {:name  => @action.description.gsub(/'/, "\\\\'"),
-                             :model => ui_lookup(:model => 'MiqAction')}
-                          end
+                          options = {:name  => @action.description.gsub(/'/, "\\\\'")}
+                          @edit ? _("Editing Action \"%{name}\"") % options : _("Action \"%{name}\"") % options
                         end
     when 'ap'
       presenter.update(:main_div, r[:partial => 'alert_profile_details', :locals => {:read_only => true}])
       right_cell_text = if @alert_profile.id.blank?
-                          _("Adding a new %{record}") % {:record => ui_lookup(:model => 'MiqAlertSet')}
+                          _("Adding a new Alert Profile")
                         else
-                          @edit ? _("Editing %{model} \"%{name}\"") % {:name => @alert_profile.description.gsub(/'/, "\\'"), :model => "#{ui_lookup(:model => @edit[:new][:mode])} #{ui_lookup(:model => 'MiqAlertSet')}"} :
-                                  _("%{model} \"%{name}\"") % {:name => @alert_profile.description.gsub(/'/, "\\'"), :model => ui_lookup(:model => 'MiqAlertSet')}
+                          name = @alert_profile.description.gsub(/'/, "\\'")
+                          if @edit
+                            _("Editing %{set_type} Alert Profile \"%{name}\"") % {
+                              :set_type => ui_lookup(:model => @edit[:new][:mode]),
+                              :name     => name
+                            }
+                          else
+                            _("Alert Profile \"%{name}\"") % {:name => name}
+                          end
                         end
     when 'al'
       presenter.update(:main_div, r[:partial => 'alert_details', :locals => {:read_only => true}])
       right_cell_text = if @alert.id.blank?
-                          _("Adding a new %{alerts}") % {:alerts => ui_lookup(:model => 'MiqAlert')}
+                          _("Adding a new Alert")
                         else
-                          pfx = @assign ? ' assignments for ' : ''
-                          msg = @edit ? _("Editing %{model} \"%{name}\"") : _("%{model} \"%{name}\"")
-                          msg % {:name => @alert.description.gsub(/'/, "\\\\'"), :model => "#{pfx} #{ui_lookup(:model => "MiqAlert")}"}
+                          options = {:name => @alert.description.gsub(/'/, "\\\\'")}
+                          @edit ? _("Editing Alert \"%{name}\"") % options : _("Alert \"%{name}\"") % options
```

Comment:
TODO: can `@assign` be set here?
Original code produced `Editing assignments for Alert "..."` then.  But I don't see any such action in UI?!